.tekton/task.yaml

```diff
@@ -0,0 +1,292 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: test
+spec:
+  params:
+    - name: repository
+      description: the git repo
+    - name: revision
+      description: the revision
+      default: master
+    - name: apikey
+      description: the ibmcloud api key
+    - name: registry-namespace
+      description: the ibmcloud registry namespace
+    - name: docker-username
+      description: for ibm container registry
+      default: iamapikey
+    - name: docker-password
+      description: iam api key
+    - name: resource-group
+      description: resource group
+      default: default
+    - name: org
+      description: organization
+    - name: region
+      description: region
+    - name: space
+      description: space
+    - name: archive-dir
+      description: archive directory
+      default: "."
+  workspaces:
+  - name: task-pvc
+    mountPath: /artifacts
+  steps:
+    - name: run-go-unittests
+      image: docker.io/yihongwang/pipeline-base-image:1.0
```

Comment:
Thanks @yhwang. It looks good to me. I think your custom image should probably be pushed to `aipipeline` or cloud registry instead? cc @Tomcli

Reply:
Good point! One information to share: I tried to use ibm container register, but it didn't work. When using classic pipeline, I only need to provide `DOCKER_USERNAME` and `DOCKER_PASSWORD` env. But this didn't work on tekton pipeline.

Reply:
@yhwang I added you to the `aipipeline` org. you can push the final images over there

Reply:
pushed image to `aipipeline` and update the task YAML.